### DISCUSSION TITLE
ReferenceError: File is not define

### DISCUSSION THREAD

**Discussion by vishnukumar89** [vercel/next.js#71965] [Help]
### Summary

Fail on build

The error occurs while running "**yarn run build**"

```
at A (/Users/my-laptop/Desktop/Projects/talent/talent-admin-fe/.next/server/chunks/212.js:1:11171)
    at /Users/my-laptop/Desktop/talent/talent/talent-admin-fe/.next/server/chunks/703.js:1:126603
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
ReferenceError: File is not defined
    at A (/Users/my-laptop/Desktop/Projects/talent/talent-admin-fe/.next/server/chunks/212.js:1:11171)
    at /Users/my-laptop/Desktop/Projects/talent/talent-admin-fe/.next/server/chunks/703.js:1:126603
ReferenceError: File is not defined
    at A (/Users/my-laptop/Desktop/Projects/talent/talent-admin-fe/.next/server/chunks/212.js:1:11171)
    at /Users/my-laptop/Desktop/Projects/talent/talent-admin-fe/.next/server/chunks/703.js:1:126603

> Build error occurred
Error: Failed to collect page data for /talents/[id]/hr-manager/[hrManagerId]
    at /Users/my-laptop/Desktop/Projects/talent/talent-admin-fe/node_modules/next/dist/build/utils.js:1268:15 {
  type: 'Error'
}
```

### Additional information

```js
Additional information
Binaries:
  Node: 20.13.1
  npm: 10.5.2
  Yarn: 1.22.22
Relevant Packages:
  next: 14.2.5
  eslint-config-next: 14.2.5
  react: 18
  react-dom: 18
  typescript: 5
```

### Example

https://github.com/vercel/next.js/discussions/56032

**Comment by icyJoseph** [comment]
Hi,

This issue happens because the Node.js 18.x.x runtime doesn't support the File API.

For example you can just run `node` and do the following:

```console
~/dev
➜ node
Welcome to Node.js v18.18.0.
Type ".help" for more information.
> typeof File
'undefined'
>
```

So, it ought to be that you are accessing File, in an unguarded way, or in code that assumes it runs only on the browser, where File does indeed exist. Show us some code snippets to help you forward.

Or,  upgrade to 20.x.x where the File Web API is now supported.

**Comment by aniruddhaadak80** [comment]
Hi @vishnukumar89 ,

The `ReferenceError: File is not defined` occurs because the `File` API is browser-specific and isn’t available server-side in Next.js. 

### Fixes:
1. **Client-Side Only**: Wrap `File` usage in a `typeof window !== 'undefined'` check to ensure it runs only on the client.
2. **Check for Server-Side References**: Ensure `File` isn't used in server functions like `getServerSideProps`.

Review the stack trace for specific locations where `File` might be referenced server-side.

**Comment by vishnukumar89** [comment]
It's solved after changing the Node version. v18 -> v20
Thanks for the response @aniruddhaadak80 , @icyJoseph .

**Reply by icyJoseph** [reply]
Nice, that's true, the later versions, 20.x.x, of Node.js have support for it.

For some reason I had switched to 18 this morning. Good job!
